src/rest/AnalyticsEndpointCaller.ts

```diff
@@ -0,0 +1,37 @@
+import { IEndpointCaller, IEndpointCallerOptions, ISuccessResponse, IEndpointCallParameters, EndpointCaller } from './EndpointCaller';
+import { UrlUtils } from '../utils/UrlUtils';
+
+export class AnalyticsEndpointCaller implements IEndpointCaller {
+  private passthrough: EndpointCaller;
+
+  constructor(public options: IEndpointCallerOptions = {}) {
+    this.passthrough = new EndpointCaller(options);
+  }
+
+  public call<T>(params: IEndpointCallParameters): Promise<ISuccessResponse<T>> {
+    if (this.shouldSendAsBeacon(params)) {
+      this.sendBeacon(params);
+      return;
+    }
+
+    return this.passthrough.call(params);
+  }
+
+  private get beaconApiIsUsable() {
+    return typeof navigator.sendBeacon === 'function';
+  }
+
+  private sendBeacon(params: IEndpointCallParameters) {
+    const url = UrlUtils.normalizeAsString({
+      paths: params.url,
+      queryAsString: params.queryString
+    });
+    const data = EndpointCaller.convertJsonToFormBody(params.requestData);
+
+    navigator.sendBeacon(url, new Blob([data], { type: 'application/x-www-form-urlencoded' }));
+  }
+
+  private shouldSendAsBeacon(params: IEndpointCallParameters): boolean {
+    return params.url.indexOf('/click') != -1 && this.beaconApiIsUsable;
```

Comment:
Why is `/click` in the url a necessary condition?

Reply:
We want to use sendBeacon only for click event, since they are the one that usually cause a page unload.